make/modules/java.management/Lib.gmk

```diff
@@ -31,15 +31,9 @@ include LibCommon.gmk
 ## Build libmanagement
 ################################################################################
 
-LIBMANAGEMENT_OPTIMIZATION := HIGH
-ifeq ($(call isTargetOs, linux)+$(COMPILE_WITH_DEBUG_SYMBOLS), true+true)
```

Comment:
On removal of "ifeq ($(call isTargetOs, linux)+$(COMPILE_WITH_DEBUG_SYMBOLS), true+true)"
..are we saying this is redundant?

It reads like Linux builds with LOW, and this change will change that to HIGH ?

I tested existing build and see -O2 in Linux fastdebug and release builds.  So this ifeq wasn't doing anything?

Windows fastdebug and release I just checked and saw -O1, I'm not sure why that is.

We do the same thing in make/modules/jdk.management/Lib.gmk so both these management locations should probably be treated the same.

(The same comparison is in make/modules/java.base/lib/CoreLibraries.gmk affecting LIBVERIFY_OPTIMIZATION, but no need to expand this change beyond the management area.)

Reply:
> Windows fastdebug and release I just checked and saw -O1, I'm not sure why that is.

The change touches only Linux so Windows stays as it is.

> We do the same thing in make/modules/jdk.management/Lib.gmk so both these management locations should probably be treated the same.

I look into this.

Reply:
@kevinjwalls 
> On removal of `ifeq ($(call isTargetOs, linux)+$(COMPILE_WITH_DEBUG_SYMBOLS), true+true)` ..are we saying this is redundant?
> 
> It reads like Linux builds with LOW, and this change will change that to HIGH ?

No, we are not saying it is redundant. Exactly as you say this PR will change the value from LOW to HIGH. The assumption is that this special case was created more than a decade ago, and no ill effect seems to arise when removing the exception, so it seems to be needed no more. 

This is part of Matthias ongoing effort to simplify and modernize the optimization levels of native libraries in the JDK.

Reply:
Thanks yes, I could not reconcile the "most likely not needed" statement with the implied "this will change the opt level", so the intent wasn't clear at first.  I think we're in sync now. 
 Agreed the opt level should be a good change!